### PR TITLE
added provider for Azure storage blobs

### PR DESCRIPTION
**Pull request by sabababaei**
#12

### CODE REVIEW

File: src/CSharp/EasyMicroservices.FileManager.AzureStorageBlobs/Interfaces/IAzureStorageBlobs.cs [hunk 1]
```diff
@@ -0,0 +1,32 @@
+ 
+using EasyMicroservices.FileManager.AzureStorageBlobs.Models;
+using Microsoft.AspNetCore.Http;
+
+namespace EasyMicroservices.FileManager.AzureStorageBlobs.Interfaces;
+
+public interface IAzureStorageBlobs
```
Comment (Ali-YousefiTelori) [inline src/CSharp/EasyMicroservices.FileManager.AzureStorageBlobs/Interfaces/IAzureStorageBlobs.cs]: We don't need to implement new interface structure, we need to use core interface.

File: src/CSharp/EasyMicroservices.FileManager.AzureStorageBlobs/Models/BlobDetail.cs [hunk 2]
```diff
@@ -0,0 +1,9 @@
+namespace EasyMicroservices.FileManager.AzureStorageBlobs.Models;
+
+public class BlobDetail
```
Comment (Ali-YousefiTelori) [inline src/CSharp/EasyMicroservices.FileManager.AzureStorageBlobs/Models/BlobDetail.cs]: We need to use core file and folder models.

File: src/CSharp/EasyMicroservices.FileManager.AzureStorageBlobs/Providers/AzureStorageBlobsProvider.cs [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+using Azure.Storage.Blobs.Models;
+using EasyMicroservices.FileManager.AzureStorageBlobs.Interfaces;
+using EasyMicroservices.FileManager.AzureStorageBlobs.Models;
+using EasyMicroservices.FileManager.Interfaces;
+using EasyMicroservices.FileManager.Models;
+using Microsoft.AspNetCore.Http;
+
+namespace EasyMicroservices.FileManager.AzureStorageBlobs.Providers;
+
+public class AzureStorageBlobsProvider :IAzureStorageBlobs
```
Comment (Ali-YousefiTelori) [inline src/CSharp/EasyMicroservices.FileManager.AzureStorageBlobs/Providers/AzureStorageBlobsProvider.cs]: It has to implement core interface.

**Review by Ali-YousefiTelori — CHANGES_REQUESTED** [review]
Hi @sabababaei please fallow the structure.

File: src/CSharp/EasyMicroservices.FileManager.AzureStorageBlobs/Providers/AzureStorageBlobsProvider.cs (hunk 3 above)
Comment (sabababaei) [reply src/CSharp/EasyMicroservices.FileManager.AzureStorageBlobs/Providers/AzureStorageBlobsProvider.cs]: I did it . Please check it out. Thanks.

**Review by Ali-YousefiTelori — APPROVED** [review]
@sabababaei Thanks for your contributions.

File: src/CSharp/EasyMicroservices.FileManager.AzureStorageBlobs.Tests/Providers/AzureStorageBlobsTest.cs [hunk 4]
```diff
@@ -0,0 +1,6 @@
+namespace EasyMicroservices.FileManager.AzureStorageBlobs.Tests.Providers;
+
```
Comment (mahdidelzendeh72) [inline src/CSharp/EasyMicroservices.FileManager.AzureStorageBlobs.Tests/Providers/AzureStorageBlobsTest.cs:10]: add test for AzureBlobTest

**Review by Mohsen-Yaghoubi — APPROVED** [review]

**Review by mahdidelzendeh72 — CHANGES_REQUESTED** [review]
Please Add test for Your new provider

**Review by mahdidelzendeh72 — APPROVED** [review]
Thank @sabababaei 
please add test also as son as possible

**Comment by sabababaei** [comment]
Ok Sure!

**Comment by Ali-YousefiTelori** [comment]
@sabababaei We have some conflicts in EasyMicroservices.FileManager.sln. please fix this when you want to push

**Comment by sabababaei** [comment]
#12

**Review by mahdidelzendeh72 — APPROVED** [review]
thanks @sabababaei

**Review by Ali-YousefiTelori — COMMENTED** [review]
Good job

**Comment by AshkanAbd** [comment]
@Ali-YousefiTelori @mahdidelzendeh72 
We have a big conflict in this PR. Golang codes and github workflows and actions has been deleted, please fix it.

**Pull request merged**